### PR TITLE
Partial revert of 'Removing uapVNext assets from individual packages (#29970)'

### PR DESCRIPTION
**Pull request by zamont**
This will re-baseline the packageIndex.json to mark System.IO.Pipes.AccessControl, System.Memory, System.Runtime.CompilerServices.Unsafe, and System.Threading.Tasks.Extensions as inbox to UAP vNext packages

### CODE REVIEW

**Comment by zamont** [comment]
Addresses #31619

**Comment by zamont** [comment]
Looking to hear feedback from @weshaggard and @joperezr 

cc: @tommcdon

**Comment by zamont** [comment]
@dotnet-bot help

**Comment by zamont** [comment]
@dotnet-bot test NETFX x86 Release Build

**Comment by zamont** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by zamont** [comment]
@ericstj can you please review this?

File: src/System.Data.SqlClient/src/Configurations.props [hunk 1]
```diff
@@ -12,9 +12,9 @@
       netcoreapp-Unix;
       netcoreapp-Windows_NT;
       uap10.0.16299-Windows_NT;
+      uap-Windows_NT;
```
Comment (ericstj) [inline src/System.Data.SqlClient/src/Configurations.props]: What is the reason for this change?  It looks to me like the UAP build and uap10.0.16299 build are identical, I don't see a good reason to put both in the package.

**Comment by ericstj** [comment]
Do you intend to ship any packages from this branch?  I don't think its versioned to do that nor would it make sense to do so codebase-wise since I imagine this branch is missing things from 2.2 and master.

**Comment by zamont** [comment]
We intend to build and ship Microsoft.Private.CoreFx.UAP so that we can update the UWP metapackage for our coming UWP 6.2 release. UWP 6.0 and 6.1 were locked to CoreFx 2.0 and this is our attempt to update to CoreFx 2.1.

**Comment by zamont** [comment]
Since we are only shipping the Microsoft.Private.CoreFX.UAP package, do I actually need any changes besides the PackageIndex.json change? That one is definitely needed to specify that we are in-boxing System.Memory and others.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 2]
```diff
@@ -2236,6 +2236,9 @@
         "4.5.1"
       ],
       "BaselineVersion": "4.4.0",
+      "InboxOn": {
+        "uap10.0.16300": "4.0.3.0"
```
[the remark this answers is not in the document]
Comment (zamont) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Yes that's the actual TFM. It's the three-part Windows SDK number.

**Comment by zamont** [comment]
I've removed all of the changes except for the packageIndex.json, since we're not shipping individual packages here just the Microsoft.Private.CoreFx.UAP framework package.

**Comment by ericstj** [comment]
Sounds good.  Can you also make sure that the changes get made to master, and potentially servicing?

**Comment by zamont** [comment]
Yes I will start that, though I might need your help or Wes's help to make sure the change lands correctly.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (joperezr) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: I don't think this will be the actual TFM we should use. RS4 shipped with a version higher than 10.0.16300 and that one is still only supporting uap6.1. If we need any changes in Windows we should change the TFM number to match whatever Windows Release will contain them.
